src/Uno.UI.RemoteControl.Server.Processors/HotReload/MetadataUpdates/CompilationWorkspaceProvider.cs

```diff
@@ -228,7 +229,17 @@ public static void RegisterAssemblyLoader()
 
 			AppDomain.CurrentDomain.AssemblyResolve += (snd, e) => Load(e.Name);
 			AppDomain.CurrentDomain.TypeResolve += (snd, e) => Load(e.Name);
-			AssemblyLoadContext.Default.Resolving += (snd, e) => Load(e.FullName);
+
+			// Processors are loaded in a separate ALC in `RemoteControlServer`, which requires
+			// to load the files in same ALC to avoid invalid cross-ALC references.
+			if (AssemblyLoadContext.GetLoadContext(typeof(CompilationWorkspaceProvider).Assembly) is { } alc)
```

Comment:
I would use `Assembly.GetExecutingAssembly()` instead. Makes code more robust to refactoring and clearer.